library/src/main/java/com/chuckerteam/chucker/internal/support/RequestProcessor.kt

```diff
@@ -0,0 +1,66 @@
+package com.chuckerteam.chucker.internal.support
+
+import android.content.Context
+import com.chuckerteam.chucker.R
+import com.chuckerteam.chucker.api.ChuckerCollector
+import com.chuckerteam.chucker.internal.data.entity.HttpTransaction
+import okhttp3.Request
+import okio.Buffer
+import okio.IOException
+import kotlin.text.Charsets.UTF_8
+
+internal class RequestProcessor(
+    private val context: Context,
+    private val collector: ChuckerCollector,
+    private val maxContentLength: Long,
+) {
+    fun process(request: Request, transaction: HttpTransaction): Request {
+        processMetadata(request, transaction)
+        processBody(request, transaction)
+        collector.onRequestSent(transaction)
+        return request
+    }
```

Comment:
```suggestion
    fun process(request: Request, transaction: HttpTransaction): Request =
        request.apply {
            ....
        }
```

nit: can be rewritten with

Reply:
I don't mind changing it to `apply` but it looks strange to me. Typically I use `apply` to set properties on an applied object. Maybe `with` is better?

Reply:
> Maybe `with` is better?

Perhaps a `let` also works

Reply:
Ah, with `with` and `let` I'd have to return explicitly from lambda. Seems `run` is the best. This got unnecessarily mind-boggling. :D